packages/components/src/components/post-rating/post-rating.tsx

```diff
@@ -0,0 +1,179 @@
+import { Component, h, Event, EventEmitter, Prop, Host, State, Element } from '@stencil/core';
+import { version } from '../../../package.json';
+
+@Component({
+  tag: 'post-rating',
+  styleUrl: 'post-rating.scss',
+  shadow: true,
+})
+export class PostRating {
+  @Element() host: HTMLPostRatingElement;
+  /**
+   * The current rating value
+   */
+  @Prop({ mutable: true }) currentRating = 0;
+
+  /**
+   * The index of the currently hovered star
+   */
+  @State() hoveredIndex: number;
+
+  /**
+   * The number of stars in the rating
+   */
+  @Prop() readonly max?: number = 5;
```

Comment:
Misleading prop name. As you mention in the JSDocs comment, this is the number (or amount) of renderd stars, not the max count of'em. You could call it `stars`, `amount`, `count`, etc.
And please define some min (3) and max (10) values for it.
https://storybook.js.org/docs/api/arg-types#controlmax

Also move this prop up and implement it as the first prop, because it is the most common prop to use on this web-component.